File: benchmark/c/cpw/sum_k_th_group_odd_positive_numbers/sum_k_th_group_odd_positive_numbers.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int sum_k_th_group_odd_positive_numbers ( int k ) ;
int sum_k_th_group_odd_positive_numbers ( int k ) {
  int cur = ( k * ( k - 1 ) ) + 1;
  int sum = 0;
  while ( k -- ) {
    sum += cur;
    cur += 2;
  }
  return sum;
}

File: translation_result/else_bms/fail/c/cpw/sum_k_th_group_odd_positive_numbers/gemini/CAPR/sum_k_th_group_odd_positive_numbers.rs

fn sum_k_th_group_odd_positive_numbers(mut k: i32) -> i32 {
    let mut cur = (k * (k - 1)) + 1;
    let mut sum = 0;
    while k > 0 {
        sum += cur;
        cur += 2;
        k -= 1;
    }
    sum
}
